newIDE/app/src/MainFrame/index.js

```diff
@@ -196,76 +201,125 @@ class MainFrame extends React.Component<Props, State> {
     gdjsDevelopmentWatcherEnabled: false,
     isPreviewFirstSceneOverriden: false,
     previewFirstSceneName: '',
-  };
-  toolbar = null;
-  _resourceSourceDialogs = {};
-  _previewLauncher: ?PreviewLauncherInterface = null;
-
-  componentWillMount() {
-    if (!this.props.integratedEditor) this.openStartPage();
-  }
-
-  componentDidMount() {
-    GD_STARTUP_TIMES.push(['MainFrameComponentDidMount', performance.now()]);
-
-    const { initialFileMetadataToOpen } = this.props;
-
-    this._loadExtensions()
-      .catch(() => {
-        /* Ignore errors */
-      })
-      .then(() => {
-        // Enable the GDJS development watcher *after* the extensions are loaded,
-        // to avoid the watcher interfering with the extension loading (by updating GDJS,
-        // which could lead in the extension loading failing for some extensions as file
-        // are removed/copied).
-        this.setState({
-          gdjsDevelopmentWatcherEnabled: true,
-        });
-      });
-    if (initialFileMetadataToOpen) {
-      this._openInitialFileMetadata(/* isAfterUserInteraction= */ false);
-    } else if (this.props.introDialog && !Window.isDev())
-      this._openIntroDialog(true);
-
-    GD_STARTUP_TIMES.push([
-      'MainFrameComponentDidMountFinished',
-      performance.now(),
-    ]);
-    console.info('Startup times:', getStartupTimesSummary());
-  }
+  });
+  const toolbar = React.useRef(null);
+  const _resourceSourceDialogs = React.useRef({});
+  const _previewLauncher = React.useRef((null: ?PreviewLauncherInterface));
+  //const { addRecentFile } = React.useContext(PreferencesContext);
+  const forceUpdate = useForceUpdate();
+
+  React.useEffect(() => {
+    // This is just for testing, to check if we're getting the right state
+    // and gives us an idea about the number of re-renders.
+    console.log(state);
+  });
+
+  const _openIntroDialog = React.useCallback(
+    (open: boolean = true) => {
+      setState(state => ({ ...state, introDialogOpen: open }));
+    },
+    [setState]
+  );
+
+  const { integratedEditor, initialFileMetadataToOpen, introDialog } = props;
+  React.useEffect(
+    () => {
+      (() => {
```

Comment:
@HarsimranVirk Is there any reason for this extra function that is self executing? I think this could be removed?

Reply:
@4ian I don't remember exactly, but I guess this has to do with the whole asynchronous code inside useEffect stuff. Im not sure, but if we want to do async stuff inside useEffect, we have to create a function and call it (or it can be a self-calling function as in this case). 

Also, setState is a bit deceiving here, it looks like the normal setState but it's not. If we don't use a self-calling function, we'll basically writing async code directly inside the useEffect, which is not recommended I believe.

Reply:
>  which is not recommended I believe.

I think it's indeed not recommended, but we don't have an async function here :) 
There is no `async () => { ... }` from what I see. Surely something that exist because you made some tries with async functions before?

We do have promises, but it's no problem at all to run them in an effect.

The setState is indeed a "set state with promise", I did not see that the return value was indeed passed. Let's keep state then.

Reply:
(so let's keep the state argument, but let's remove the self-calling function because it should not be useful as we don't use async/await - unless I missed something else!)

Reply:
> Surely something that exist because you made some tries with async functions before?

Yes, now that you've mentioned it, I remember this was an async function :) And I had to remove async functions because of the whole babel transpiling issue :)

Alright, so I guess it's not necessary now. I'll remove it and test it then :+1:

Reply:
Sounds good 👍